File: app/src/main/java/net/feheren_fekete/applist/applistpage/viewmodel/AppShortcutItem.java
package net.feheren_fekete.applist.applistpage.viewmodel;

public class AppShortcutItem extends StartableItem {

    private String mName;
    private String mPackageName;
    private String mShortcutId;
    private String mIconPath;

    public AppShortcutItem(long id, String name, String packageName, String shortcutId, String iconPath) {
        super(id);
        mName = name;
        mPackageName = packageName;
        mShortcutId = shortcutId;
        mIconPath = iconPath;
    }

    @Override
    public String getName() {
        return mName;
    }

    public String getPackageName() {
        return mPackageName;
    }

    public String getShortcutId() {
        return mShortcutId;
    }

    public String getIconPath() {
        return mIconPath;
    }

}

File: app/src/main/java/net/feheren_fekete/applist/applistpage/viewmodel/AppShortcutItem.kt
package net.feheren_fekete.applist.applistpage.viewmodel

class AppShortcutItem(id: Long,
                      name: String,
                      customName: String,
                      customIconPath: String,
                      val packageName: String,
                      val shortcutId: String,
                      val iconPath: String,
                      parentSectionId: Long) : StartableItem(id, name, customName, customIconPath, parentSectionId)

